swarm/storage/mru/topic.go

```diff
@@ -0,0 +1,56 @@
+package mru
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/ethereum/go-ethereum/common/bitutil"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/swarm/storage"
+)
+
+const topicLength = storage.KeyLength
+
+type Topic struct {
```

Comment:
Why does it need to be a `struct`?

Reply:
To avoid people hacking into it and treating it as an array. Topic is backed by an array now, but this could change.